contracts/Strategy.sol

```diff
@@ -109,59 +99,56 @@ contract Strategy is BaseStrategy {
     }
 
     function _initializeStrat(
-        uint256 _maxSingleInvest,
-        uint256 _minTimePerInvest,
-        uint256 _slippageProtectionIn,
         address _curvePool,
         address _curveToken,
         address _yvToken,
         uint256 _poolSize,
         bool _hasUnderlying
     ) internal {
-        require(want_decimals == 0, "Already Initialized");
+        require(synth_decimals == 0, "Already Initialized");
         require(_poolSize > 1 && _poolSize < 5, "incorrect pool size");
+        require(address(want) == address(_synthsUSD()), "want must be sUSD");
```

Comment:
I would kill both require msgs to reduce contract size.

Reply:
I am leaving it until we hit contract size limit, to avoid human errors